docs/developer/local-docker.md

```diff
@@ -50,6 +59,97 @@ This will launch the integration test  runner against the CI environment, using
 
 This may all be overridden.
 
+## Unit tests
+
+Unit tests require a running Narrative server in order to access Jupyter javascript files which are only available when the server is built and running.
+
+As a server, it can be accessed within a container is well, and for those who prefer Docker-based workflows, is quite convenient.
+
+- run the container as specified above. For example:
+
+```shell
+make dev-image
+ENV=ci make run-dev-image
+```
+
+Then, to run tests without authentication.
+
+```shell
+nvm use 16
+npm install
+NARRATIVE_SERVER_URL=http://localhost:8888 npm run test_local
+```
+
+To run with authentication, follow the normal instructions, but briefly:
+
+- obtain a KBase token for a given user, such as yourself. This token does not require any special privileges.
+- in `test/testConfig.json` you may change `SOMETESTUSER` to the associated username, or leave as is
+- create `test/SOMETESTUSER.tok` (or using the actual username as changed above) and make the contents the token string.
+
+### Other conveniences
+
+The unit tests, by default, are quite noisy. To quiet them down make the following changes to the test configuration:
+
+(all of these changes may not be necessary, but the combination of them does inhibit all logging, with just test results left)
+
+- in `test/unit/karma.conf.js` change `logLevel: config.LOG_INFO,` to `logLevel: config.LOG_DISABLE,`
+
+after the line:
+
+```javascript
+    config.reporters = ['kjhtml', 'brief'];
+```
+
+add this:
+
+``` javascript
+    config.briefReporter = {
+        suppressBrowserLogs: true
+    }
+```
+
+Now your tests, if they pass, will be quite pleasant to gaze at:
+
+```shell
+erikpearson@Eriks-MacBook-Pro narrative % NARRATIVE_SERVER_URL=http://localhost:8888 npm run test_local
+
+> kbase-narrative-core@5.1.3 test_local /Users/erikpearson/Work/KBase/2022/helpdesk/PTV-1620/pr2redux/narrative
+> karma start test/unit/karma.local.conf.js
+
+
+Chrome Headless 108.0.5359.124 (Mac OS 10.15.7) WARN: 'Loaded token file test/SOMETESTUSER.tok'
+TOTAL: 13 SUCCESS 13 passed      0 failed      0 skipped            
+   13 total       13 passed      0 failed      0 skipped 
+```
+
+To focus tests, you can modify `karma.local.conf.js`. Here we have focussed on the tests that
+are directly contained within `test/unit/spec/widgets/*`, which at the moment, is just `kbaseTabs-spec.js`.
```

Comment:
You might want to add comments about how to do this using the native jasmine API -- edit the test files to use `fdescribe` or `fit` in place of `describe` or `it` respectively (and `xdescribe` and `xit` to skip tests).

Reply:
Good point. I just wanted to point out this other technique for focusing/excluding at the directory level.

And I do appreciate that you actually read this humble little doc and suggested improvements.